### PR TITLE
Sockets to notify facilitators of students completing quiz

### PR DESCRIPTION
**Pull request by coati-00**
I removed the code for notifying students and decided to do it on a new branch, this pr contains the code to notify a facilitators when a student is at a decision block or gateblock

### CODE REVIEW

File: uelc/main/views.py [hunk 1]
```diff
@@ -146,6 +153,20 @@ def get_library_items(self, case):
         library_items = LibraryItem.objects.filter(case=case, user=user)
         return library_items
 
+    def notify_fascilitators(self, request, path, notification):
```
Comment (thraxil) [inline uelc/main/views.py]: "facilitators", not "fascilitators".

File: uelc/main/views.py [hunk 2]
```diff
@@ -146,6 +153,20 @@ def get_library_items(self, case):
...
+        socket.connect(settings.WINDSOCK_BROKER_URL)
+        msg = dict(user_id=user.id,
+                   path=path,
+                   section_pk=self.section.pk,
+                   notification=notification)
+        e = dict(address="%s.pages/%s/facilitator/" %
+                 (settings.ZMQ_APPNAME, self.section.hierarchy.name),
+                 content=json.dumps(msg))
+        socket.send(json.dumps(e))
+        socket.recv()
```
Comment (thraxil) [inline uelc/main/views.py]: Yep. Looks like you figured out how to talk to windsock.

**Comment by thraxil** [comment]
I'm going to offer some suggestions on improving your git workflow. This is a fairly substantial feature and I'm glad it looks like you've basically got it, but you may notice that most other pull requests don't have this many commits and travis failures.

First, I would suggest that you make a very strong habit of not doing a `git push` without running `make` immediately beforehand. `make` runs all the tests that Travis is going to and will tell you beforehand if Travis is going to fail it. It's always going to be better to fix issues locally before sharing the code. Personally, I never even type a bare `git push ...`, I instead *always* type `make && git push ...`. That makes it basically nearly impossible for me to push code that Travis will fail.

If the make step fails on flake8 or jshint or something, you can fix it and, since you haven't pushed the code, you can just do `git commit -a --amend -C HEAD` instead of a plain commit and it will make your fixes go in as if they were in the original commit. Any time you find yourself not able to come up with a more descriptive commit message than `update`, that's almost certainly what you want to do. The result will be a much cleaner git history that's easier for us to review. As long as you haven't pushed the commits anywhere, you can ammend them together and keep things much tidier. I set up a git alias so I only have to type 'git caa' because I use that command so often. (there's also a whole cherry picking process that you can do to clean up history and merge commits together, but this is the simple approach to start with).

**Comment by coati-00** [comment]
Oh okay sorry

**Comment by coati-00** [comment]
hmmmm when I run make locally it passes, it doesn't seem to be running ./node_modules/jscs/bin/jscs media/js/uelc_admin.... when I run that on its own all of these errors come up

**Pull request merged**